index.js

```diff
@@ -4,16 +4,25 @@ var Cache = require('hashlru')
 var inject = require('./inject')
 function id (e) { return e }
 function isNumber(n) { return 'number' == typeof n && !isNaN(n) }
-module.exports = function (file, block_size, codec) {
-  if(!isNumber(block_size))
-    codec = block_size, block_size = 1024*16
-  codec = codec || {encode: id, decode: id}
 
-  var blocks = Blocks(file, block_size,'a+', Cache(1024))
-  return inject(
-    blocks,
-    createFrame(blocks, codec),
-    codec,
-    file
-  )
+module.exports = function (opts) {
+  if (typeof opts !== 'object') opts = legacy(arguments[0], arguments[1], arguments[2])
+
+  var file = opts.file
+  var blockSize = opts.blockSize || 1024*16
+  var codec = opts.codec || {encode: id, decode: id}
+  var flags = opts.flags || 'a+'
+  var cache = opts.cache || Cache(1024)
```

Comment:
We could also remove the cache fallback as there is one in the `aligned-block-file` module